### PR TITLE
Chore/sc 37018 organic uploader tests

### PR DESCRIPTION
**Pull request by danielcastro-studocu**
In this branch I’m adding more scenarios for the organic uploader, refactoring other pages to adopt the encapsulation approach suggested by @Tatiinzhee, moving functions to other pages according to the scope (like moving `clickloginButton` to `header.ts`).

### CODE REVIEW

File: framework/pages/registration.ts [hunk 1]
```diff
@@ -25,9 +25,13 @@ const clickRegisterButton = async (page: Page, isMobile: Boolean) => {
...
 
 const RegistrationPage = {
+    clickRegisterButton: async (page: Page, isMobile: any) => {
+        await clickRegisterButton(page, isMobile);
+    },
+
     registerWithEmail:async (page: Page, firstname: string, lastmame: string, emailaddress: string, password: string) => {
-        await openLandingPage(page);
-        await closeCookiesDisclaimer(page);
+        await app().LandingPage().openLandingPage(page);
```
Comment (mila-qa) [inline framework/pages/registration.ts:33]:
```suggestion
        await app().Landing().openLandingPage(page);
```

File: tests/uploader.spec.ts [hunk 2]
```diff
@@ -1,16 +1,59 @@
...
+    await app().Uploader().uploadAndSubmitDocument(page);
+    await app().Uploader().selectCourse(page);
+    await app().Uploader().fillInDocumentDetails(page, "Summaries", undefined, "Book", "Organische Chemie");
+    await page.waitForURL('**/upload/complete**', {timeout: 30000});
+    await waitForVisible(page, "14-premium-days-received");
+});
+
+test(`Guest uploader page is prefilled with university`, async ({page, isMobile}, testInfo) => {
+    // University is not prefilled in Webkit and not changing on Mobile
+    if(testInfo.project.name == 'webkit' || isMobile){
```
Comment (mila-qa) [inline tests/uploader.spec.ts:39]: Daniel, just out of curiosity, what is wrong with webkit?

**Review by mila-qa — COMMENTED** [review]
Solid and clean 💪. Thank you Daniel!

File: tests/uploader.spec.ts (hunk 2 above)
Comment (danielcastro-studocu) [reply tests/uploader.spec.ts:40]: Maybe it’s a specific situation that is happening only in my localenv, but even when I run it manually the university that is selected in the Search step is not pre-filled in the uploader details form. There’s also an error in the pre-filled Title which should be related to the filename.
I’m submitting the code as it is for now and planning to confirm if it’s a real bug in other local environments or as soon as my localenv is more stable and refactor this code as soon as I confirm that it’s working fine.

File: framework/pages/uploader.ts [hunk 3]
```diff
@@ -1,79 +1,264 @@
...
+    allowSingleUploads: async (page: Page) => {
+        const currentURL = page.url();
+        await page.goto(currentURL + '?allowSingleUploads=true');
+        await waitForNotVisible(page, minimumAllowedDocumentsText);
+    },
+    
+    uploadAndSubmitDocument: async (page: Page, filename: string = 'Atomic_number.pdf') => {
+        await page.setInputFiles(inputFile, filename);
+        await waitForVisible(page, uploadedDocumentName);
+        await clickOnElement(page, submitDocumentsButton);
```
[the remark this answers is not in the document]
Comment (mila-qa) [reply framework/pages/uploader.ts:157]: Last update: `Upload document and see premium days` seems to be an unstable test. It's not related to the local pipeline issue. After chatting with @danielcastro-studocu we figured out 2 moments:
  - clickOnElement() should have element visibility check;
  - setInputFiles() for a flow with logged-in user doesn't work stable and requires further investigation. Found temporarily workaround
  
 ```
  uploadAndSubmitDocument: async (page: Page, filename: string = 'Atomic_number.pdf') => {
        await page.waitForTimeout(3000);
        await page.setInputFiles(inputFile, filename);
        ...
        }
```

File: tests/uploader.spec.ts [hunk 4]
```diff
@@ -1,15 +1,60 @@
...
+    const userPwd = process.env.PASSWORD;
+    
+    test.setTimeout(120000);
+    await app().Landing().openLandingPage(page);
+    await app().CookieBanner().submitCookieBanner(page);
+    await app().Header().clickloginButton(page, isMobile);
+    await app().Authentication().loginWithEmail(page, userName, userPwd);
+    await app().Uploader().openOrganicUploader(page);
+    await app().Uploader().allowSingleUploads(page);
+    await app().Uploader().uploadAndSubmitDocument(page);
```
Comment (mila-qa) [inline tests/uploader.spec.ts:31]: Test is failing on the below step in 90% of runs
`Running 1 test using 1 worker
  1) [chromium] › tests/uploader.spec.ts:20:6 › Upload document and see premium days ===============

    Test timeout of 120000ms exceeded.

    locator.waitFor: Target closed
    =========================== logs ===========================
    waiting for selector "[data-test-selector="uploaded-document-name"] >> nth=0" to be visible
    ============================================================

      29 |     await app().Uploader().openOrganicUploader(page);
      30 |     await app().Uploader().allowSingleUploads(page);
    > 31 |     await app().Uploader().uploadAndSubmitDocument(page);
         |     ^
`

**Review by mila-qa — CHANGES_REQUESTED** [review]

File: tests/uploader.spec.ts (hunk 4 above)
Comment (mila-qa) [reply tests/uploader.spec.ts:30]: in  https://github.com/StuDocu/playwright-tests-sandbox/pull/9/commits/4f58a16972e075628f8ddea52059e451041ae478 and https://github.com/StuDocu/playwright-tests-sandbox/pull/9/commits/3b8d8fc94f9c3c2686be761237be773e468a29a9

**Review by mila-qa — APPROVED** [review]

**Review by prachistudocu — APPROVED** [review]

**Pull request merged**